src/Powercord/plugins/pc-emojiUtility/index.js

```diff
@@ -1,42 +1,173 @@
 const Plugin = require('powercord/Plugin');
-const { getModule } = require('powercord/webpack');
+const { getModule, channels: { getSelectedChannelState, getChannelId }, constants: { Routes, APP_URL_PREFIX }, React, messages: { createBotMessage, receiveMessage } } = require('powercord/webpack');
+
+const { existsSync, createWriteStream } = require('fs');
+const { get } = require('https');
+const { extname } = require('path');
+const { parse } = require('url');
+
 const emojiStore = getModule([ 'getGuildEmoji' ]);
 
+const Settings = require('./Settings.jsx');
+
 module.exports = class EmojiUtility extends Plugin {
   getEmojiRegex () {
     return /^<a?:([a-zA-Z0-9_]+):([0-9]+)>$/;
   }
 
+  getGuildUrl (guild) {
+    let selectedChannel = getSelectedChannelState()[guild.id];
+    if (!selectedChannel) {
+      /* I am not sure if it can get here but just to be sure */
+      selectedChannel = guild.systemChannelId;
+    }
+
+    return APP_URL_PREFIX + Routes.CHANNEL(guild.id, selectedChannel); // eslint-disable-line new-cap
+  }
+
+  getFullEmoji (emoji) {
+    return `<${(emoji.animated ? 'a' : '')}:${emoji.name}:${emoji.id}>`;
+  }
+
+  download (url) {
+    return new Promise((resolve, reject) => {
+      get(url)
+        .on('response', (response) => resolve(response))
+        .on('error', (error) => reject(error));
+    });
+  }
+
+  write (path, stream) {
+    return new Promise((resolve, reject) => {
```

Comment:
you can use `writeFile` from fs.promises, so it's just `writeFile(path, contents);`